src/library/Cardano/CoinSelection/Fee.hs

```diff
@@ -131,35 +133,75 @@ data FeeOptions i o = FeeOptions
     } deriving Generic
 
 newtype FeeAdjustmentError
-    = ErrCannotCoverFee Fee
-    -- ^ UTxO exhausted during fee covering
-    -- We record what amount missed to cover the fee
+    = CannotCoverFee Fee
+    -- ^ Indicates that the given map of additional inputs was exhausted while
+    --   attempting to select extra inputs to cover the required fee.
+    --
+    -- Records the shortfall between the required fee and the total value
+    -- of currently selected inputs.
+    --
     deriving (Show, Eq)
 
--- | Given the coin selection result from a policy run, adjust the outputs for
--- fees, potentially returning additional inputs that we need to cover all
--- fees.
+-- | Adjusts the given 'CoinSelection' in order to pay for a __transaction__
+--   __fee__, required in order to publish the selection as a transaction on
+--   a blockchain.
+--
+-- == Background
+--
+-- Implementations of 'Cardano.CoinSelection.CoinSelectionAlgorithm' generally
+-- produce coin selections that are /exactly balanced/, satisfying the
+-- following equality:
+--
+-- >>> sumInputs c = sumOutputs c + sumChange c
+--
+-- In order to pay for a transaction fee, the above equality must be
+-- transformed into an /inequality/:
+--
+-- >>> sumInputs c > sumOutputs c + sumChange c
+--
+-- The difference between these two sides represents value to be paid /by the/
+-- /originator/ of the transaction, in the form of a fee:
```

Comment:
I remember in the days of `cardano-sl` that there was a distinction between situtations that sender-paying-fee and recepient-paying-fee. Probably it is not the case anymore, and sender-paying-fee is tantamount to originator. But in past days it was not true.. Maybe it would be worth double check on this

Reply:
I'm pretty sure that sender-pays-fee is the same as originator-pays-fee, just phrased differently.

Perhaps we could rename "originator" to "sender" to make this clearer?

Reply:
After a brief internal discussion, we decided to leave the description as it is. :+1: